app/controllers/navigation/Navigator.scala

```diff
@@ -46,14 +46,19 @@ class Navigator @Inject()(appConfig: AppConfig, auditService: AuditService) {
     }
 
   def continueTo(mode: Mode, factory: Mode => Call)(implicit req: JourneyRequest[AnyContent], hc: HeaderCarrier): Result =
-    redirectTo(mode.next, factory)
+    FormAction.bindFromRequest match {
+      case SaveAndReturn =>
+        auditService.auditAllPagesUserInput(AuditTypes.SaveAndReturnSubmission, req.cacheModel)
+        goToDraftConfirmation()
+      case _ => Results.Redirect(factory(mode.next))
+    }
 
-  def redirectTo(mode: Mode, factory: Mode => Call)(implicit req: JourneyRequest[AnyContent], hc: HeaderCarrier): Result =
+  def continueTo(mode: Mode, itemId: String, factory: (Mode, String) => Call)(implicit req: JourneyRequest[AnyContent], hc: HeaderCarrier): Result =
```

Comment:
What is the usage of this itemId?

When and where you want to use it?

Reply:
I think it might be better to do this way:
`def continueTo(mode: Mode, itemId: ItemId, factory: (Mode, ItemId) => Call)(implicit req: JourneyRequest[AnyContent], hc: HeaderCarrier): Result =`

Reply:
Yeah, but this method is handling the non items pages too. The question was why now the `itemId` is the mandatory for this method